### PR TITLE
Refactor warehouse mutations

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Deprecate the `shippingZone` from the `WarehouseCreateInput`
- Add shipping zones validation in `WarehouseShippingZoneAssign`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7702 Add shipping zone validation in  warehouse shipping zone assignment mutations](https://app.clickup.com/t/2549495/SALEOR-7702)

File: saleor/graphql/warehouse/mutations.py [hunk 1]
```diff
@@ -48,12 +51,8 @@ def clean_input(cls, info, instance, data, input_cls=None):
...
                 raise ValidationError({"name": error})
 
-        shipping_zones = cleaned_input.get("shipping_zones", [])
-        if not validate_warehouse_count(shipping_zones, instance):
-            msg = "Shipping zone can be assigned only to one warehouse."
-            raise ValidationError(
-                {"shipping_zones": msg}, code=WarehouseErrorCode.INVALID
-            )
+        # assigning shipping zones in the WarehouseCreate mutation is deprecated
+        cleaned_input.pop("shipping_zones", None)
```
Comment (korycins) [inline saleor/graphql/warehouse/mutations.py:55]: Isn't it a breaking change? We leave an input but we change the behaviour of using it.

File: saleor/graphql/warehouse/mutations.py [hunk 2]
```diff
@@ -126,11 +125,88 @@ def perform_mutation(cls, _root, info, **data):
...
+
+        Raise and error when the condition is not fulfilled.
+        """
+        shipping_zone_ids = [zone.id for zone in shipping_zones]
+        ChannelShippingZone = channel_models.Channel.shipping_zones.through
+        channel_shipping_zones = ChannelShippingZone.objects.filter(
+            shippingzone_id__in=shipping_zone_ids
+        )
+
+        # shipping zone cannot be assigned when aby channel is assigned to the zone
```
Comment (korycins) [inline saleor/graphql/warehouse/mutations.py:154]: `when aby channel`?

File: saleor/graphql/warehouse/mutations.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/warehouse/mutations.py:55]: I do not think so, it's a fix for the feature that was introduced in 3.5. And it shouldn't be possible of doing such a thing. It should be blocked in 3.5.

Comment (maarcingebala) [reply saleor/graphql/warehouse/mutations.py:55]: We should mention this as a breaking change in the changelog, this behavior was here for since many versions back.

Comment (patrys) [reply saleor/graphql/warehouse/mutations.py:55]: Can we at least have it reject the input and not ignore it silently?

Comment (IKarbowiak) [reply saleor/graphql/warehouse/mutations.py]: Done.

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/core/types/common.py [hunk 3]
```diff
@@ -339,7 +339,7 @@ class PaymentError(Error):
     code = PaymentErrorCode(description="The error code.", required=True)
     variants = NonNullList(
         graphene.ID,
-        description="List of varint IDs which causes the error.",
```
Comment (fowczarek) [inline saleor/graphql/core/types/common.py:342]: The unnecessary change. We have `variant IDs` here.

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/core/types/common.py (hunk 3 above)
Comment (maarcingebala) [reply saleor/graphql/core/types/common.py:342]: It's a typo fix
